instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/AgentContextStorage.java

```diff
@@ -40,6 +40,11 @@
   private static final Logger logger = LoggerFactory.getLogger(AgentContextStorage.class);
 
   public static final AgentContextStorage INSTANCE = new AgentContextStorage();
```

Comment:
Is it possible to use the extension?

https://github.com/open-telemetry/opentelemetry-java/blob/main/context/src/main/java/io/opentelemetry/context/ContextStorage.java#L112

Does it break muzzle?

Reply:
Yes, it fails muzzle when older api is used because I call `ContextStorage.root()` which was just added. Seems like wrapper based approach won't work on older api because `Context.root()` just returns `ArrayBasedContext.root()` which we would still need to wrap. Similarly old approach would fail when `ArrayBasedContext` is not used, should also wrap what is returned from `ContextStorage.root()`.

Reply:
Replaced call to `ContextStorage.root()` with method handle, @anuraaga could you review.